vaadin-grid-flow-parent/vaadin-grid-flow/src/main/java/com/vaadin/flow/component/grid/Grid.java

```diff
@@ -415,6 +416,40 @@ public enum MultiSortPriority {
         PREPEND
     }
 
+    /**
+     * Sets the default multi-sort priority to use for all Grid instances.
+     * <p>
+     * This method should be called before creating any Grid instances. Changing
+     * this setting does not affect the default for existing Grids. Use
+     * {@link Grid#setMultiSort(boolean, MultiSortPriority)} to provide a custom
+     * multi-sort priority overriding the default priority for a single Grid.
+     * <p>
+     * This method may only be invoked inside of UI threads, and will throw
+     * otherwise. This setting being scoped to the current UI means that it will
+     * stay active when navigating between pages using the Vaadin router, but
+     * not when doing a "hard" location change, or when reloading the page. As
+     * such it is recommended to apply this setting on every page that displays
+     * a Grid. Note that when using the preserve on refresh feature, a view's
+     * constructor is not called. In that case this setting can be applied in an
+     * attach listener.
+     *
+     * @param priority
+     *            the multi-sort priority to be used by all grid instances
+     */
+    public static void setDefaultMultiSortPriority(MultiSortPriority priority) {
+        UI ui = UI.getCurrent();
+        if (ui == null || ui.getPage() == null) {
+            throw new IllegalStateException("UI instance is not available. "
+                    + "It means that you are calling this method "
+                    + "out of a normal workflow where it's always implicitly set. "
+                    + "That may happen if you call the method from the custom thread without "
+                    + "'UI::access' or from tests without proper initialization.");
+        }
+        UI.getCurrent().getPage().executeJs(
+                "window.Vaadin.Flow.gridConnector.setDefaultMultiSortPriority($0)",
```

Comment:
For the Map component we needed the `executeJs` call, and the UI scope constraints, because the state is effectively stored in a JS library running in the browser page, and we wanted to be able to differentiate per UI / per user.

However for this use-case, I feel we could make this a lot simpler by making the setting global by storing the default in a static property, and then applying it in the `Grid` constructor by setting the respective attribute. Using the `setDefaultMultiSortPriorityAppend` function of the web component does not seem necessary here.

Reply:
Good point, thanks. Updated as you suggested and reverted the change to the connector.